tests/test_macros.py

```diff
@@ -0,0 +1,92 @@
+import dataclasses
+import os
+
+import dbt.tracking
+import pytest
+from _pytest.fixtures import SubRequest
+from dbt.adapters.factory import get_adapter, register_adapter, AdapterContainer
+from dbt.clients.jinja import MacroGenerator
+from dbt.config.runtime import RuntimeConfig
+from dbt.context import providers
+from dbt.contracts.connection import ConnectionState
+from dbt.contracts.graph.manifest import Manifest
+from dbt.adapters.spark.connections import (
+    SparkConnectionManager,
+    PyodbcConnectionWrapper,
+)
+from dbt.parser.manifest import ManifestLoader
+from dbt.tracking import User
+from pyspark.sql import SparkSession
+from sodaspark.scan import Connection
```

Comment:
See the [issue](https://github.com/dbt-labs/dbt-spark/issues/272) in `dbt-spark` about replacing the `Connection` with a reference to the `SparkSession`